crates/lib/src/executable.rs

```diff
@@ -478,6 +482,7 @@ impl<'execution> Executable<'_, 'execution> {
         &mut self,
         quantum_processor_id: S,
         translation_options: Option<TranslationOptions>,
+        connection_strategy: ConnectionStrategy,
```

Comment:
I would pool these into an `ExecutionOptions` with a builder:

* This'll be the last time we have to make a breaking change in this signature. Future options can just be added to that struct
* Keeps this from becoming a longer and longer list as we add more configuration

Reply:
Something I keep tripping over as we add these options structs is that they are all constructed and modified differently. Some use public fields, some mutate self, others return updated copies, none so far use an actual builder type. What do you have in mind when you say builder? I can use that approach for `ExecutionOptions`, then create a ticket to consolidate them. Maybe [`derive_builder`](https://docs.rs/derive_builder/latest/derive_builder/)?

Reply:
Good eye! yeah that stuff matters in usage but can be hard to spot in reviews and focused work. I do like `derive_builder` and that provides a friendly UX.

Reply:
Okay, switched to using `ExecutionOptions`